### PR TITLE
Some improvements for loggy and filtering

### PR DESCRIPTION
**Pull request by lukaknezic**
I made some improvements to the lib. As I was writing article about it I noticed some things can be done better so I decided to fix them. This is mainly for filtering and whitelist/blacklist (removed them now, they are part of filters).
Some minor changes include default printer, and how loggy_flutter and loggy_flutter_dio were not using library so I updated that

### CODE REVIEW

**Review by itsJoKr — APPROVED** [review]
lgtm

**Review by Truba — APPROVED** [review]

File: loggy/lib/src/filters/whitelist_filter.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+/// White list filter will log only [Type] of loggy that is passed.
+/// You should pass only [Type] that is from [LoggyType] otherwise no effect will take place.
+class WhitelistFilter extends LoggyFilter {
+  const WhitelistFilter(this._types);
+
+  final List<Type> _types;
+
+  @override
+  bool shouldLog(LogLevel level, Type type) {
+    return _types.isEmpty || _types.contains(type);
```
Comment (zenled) [inline loggy/lib/src/filters/whitelist_filter.dart:12]: I'm a bit confused here.
The documentation is not in sync with the behavior (I think you should change the behavior).

If I understand this correctly, the `WhitelistFilter` only shows logs from `LoggyType`-s that are passed.
So if I create a `WhitelistFilter([])` nothing should be logged. 

In the example below:
The log is: `👻 08:05:46.351346 INFO     Example - abc` 
I would expect the log to be empty

```dart
void main() {
  Loggy.initLoggy(
    logPrinter: const PrettyPrinter(),
    logOptions: const LogOptions(
      LogLevel.all,
    ),
    filters: [
      WhitelistFilter([]),
    ],
  );

  MyExample();
}

class MyExample with ExampleLoggy {
  MyExample(){
    loggy.info("abc");
  }
}
```

File: loggy/lib/src/filters/blacklist_filter.dart [hunk 2]
```diff
@@ -0,0 +1,11 @@
+part of loggy;
+
+/// Black list filter, opposite of [WhitelistFilter]
+class BlacklistFilter extends WhitelistFilter {
+  BlacklistFilter(List<Type> types) : super(types);
+
+  @override
+  bool shouldLog(LogLevel level, Type type) {
+    return !super.shouldLog(level, type);
```
Comment (zenled) [inline loggy/lib/src/filters/blacklist_filter.dart:9]: This ties in with my comment for `WhitelistFilter`.

In the example below:
Nothing gets logged.
I would expect the log to be: `👻 08:05:46.351346 INFO Example - abc`

```dart
void main() {
  Loggy.initLoggy(
    logPrinter: const PrettyPrinter(),
    logOptions: const LogOptions(
      LogLevel.all,
    ),
    filters: [
      BlacklistFilter([]),
    ],
  );

  MyExample();
}

class MyExample with ExampleLoggy {
  MyExample(){
    loggy.info("abc");
  }
}
```

**Review by zenled — CHANGES_REQUESTED** [review]

File: flutter_loggy_dio/lib/flutter_loggy_dio.dart [hunk 3]
```diff
@@ -1,2 +1,10 @@
-export 'package:flutter_loggy_dio/flutter_loggy_dio/custom_loggers.dart';
-export 'package:flutter_loggy_dio/flutter_loggy_dio/dio_logger.dart';
+library loggy_flutter_dio;
```
[the remark this answers is not in the document]
Comment (lukaknezic) [reply flutter_loggy_dio/lib/flutter_loggy_dio.dart:1]: Yes, updated :D

File: loggy/lib/src/filters/whitelist_filter.dart (hunk 1 above)
Comment (lukaknezic) [reply loggy/lib/src/filters/whitelist_filter.dart:12]: Oh wow, good catch. Anyway I updated behaviour to make sense now (For whitelist and blacklist) and added tests for filters with that cases :)

**Review by zenled — APPROVED** [review]

**Pull request merged**